sdk/tools/xml2sdb/xml2sdb.cpp

```diff
@@ -17,7 +17,7 @@ static const GUID GUID_NULL = { 0 };
 static const char szCompilerVersion[] = "1.7.0.1";
 
 #if !defined(C_ASSERT)
-#define C_ASSERT(expr) extern char (*c_assert(void)) [(expr) ? 1 : -1]
+#define C_ASSERT(expr) typedef char __C_ASSERT__[(expr) ? 1 : -1]
```

Comment:
Why does this need to change?
Didn't it work before?

Reply:
mspaint uses C++. The target macro declares extern function. An extern function is either C function or C++. If they are mixed in a module, that causes linkage errors.